src/index.tsx

```diff
@@ -1,14 +1,16 @@
-import React, { createContext, useContext, useReducer, useEffect } from "react";
+import * as React from "react";
 
 import useLocalStorage from "./useLocalStorage";
 
+const { createContext, useContext, useReducer, useEffect } = React;
+
 const SET_ITEMS = "SET_ITEMS";
 const ADD_ITEM = "ADD_ITEM";
 const UPDATE_ITEM = "UPDATE_ITEM";
 const REMOVE_ITEM = "REMOVE_ITEM";
 const EMPTY_CART = "EMPTY_CART";
 
-const CartContext = createContext();
+const CartContext = createContext<any>(undefined);
 
 export const initialState = {
```

Comment:
It's probably worth us typing the `initialState` now.

Reply:
Yes, before we do that, I have a hack on my repo, that is very needed to make it work proper, this is because I want to avoid adding dummy functions to the type before it is used, like `emptyCart: () => void`

Reply:
I'll make a PR to resolve this.

Reply:
Added changes as a commit to this